atlassian_jwt_auth/verifier.py

```diff
@@ -31,15 +40,30 @@ def verify_jwt(self, a_jwt, audience, leeway=0, **requests_kwargs):
         """
         key_identifier = key._get_key_id_from_jwt_header(a_jwt)
         public_key = self._retrieve_pub_key(key_identifier, requests_kwargs)
-
+        public_key_obj = public_key
+        if not isinstance(public_key, (RSAPublicKey, EllipticCurvePublicKey)):
+            alg = jwt.get_unverified_header(a_jwt).get('alg', None)
+            public_key_obj = self._load_public_key(public_key, alg)
         return self._decode_jwt(
-            a_jwt, key_identifier, public_key,
+            a_jwt, key_identifier, public_key_obj,
             audience=audience, leeway=leeway)
 
     def _retrieve_pub_key(self, key_identifier, requests_kwargs):
         return self.public_key_retriever.retrieve(
             key_identifier, **requests_kwargs)
 
+    @lru_cache(maxsize=10)
+    def _load_public_key(self, public_key, algorithm):
+        """ Returns a public key object instance given the public key and
+            algorithm.
+        """
+        if algorithm not in self.algorithms:
+            raise InvalidAlgorithmError(
+                'The specified alg value is not allowed')
```

Comment:
Can we make the error tell the reader which algos are allowed?

Reply:
Yes, although they are part of the s2s asap spec and iirc this error matches what the pyjwt library produces.